bot/exts/info/code_snippets.py

```diff
@@ -255,16 +255,23 @@ async def on_message(self, message: discord.Message) -> None:
             except discord.NotFound:
                 # Don't send snippets if the original message was deleted.
                 return
-
-            if len(message_to_send) > 1000 and message.channel.id != Channels.bot_commands:
-                # Redirects to #bot-commands if the snippet contents are too long
-                await self.bot.wait_until_guild_available()
-                destination = self.bot.get_channel(Channels.bot_commands)
-
-                await message.channel.send(
-                    'The snippet you tried to send was too long. '
-                    f'Please see {destination.mention} for the full snippet.'
-                )
+            except discord.Forbidden as e:
+                # We still want to send snippets when in DMs, but if we're in guild then
+                # reraise error since that means there's a permissions issue with the bot.
+                if message.guild:
+                    raise e
+
+            # If we're in a guild, then check if we need to redirect to #bot-commands
+            if destination.guild:
+                if len(message_to_send) > 1000 and message.channel.id != Channels.bot_commands:
+                    # Redirects to #bot-commands if the snippet contents are too long
+                    await self.bot.wait_until_guild_available()
+                    destination = self.bot.get_channel(Channels.bot_commands)
+
+                    await message.channel.send(
+                        'The snippet you tried to send was too long. '
+                        f'Please see {destination.mention} for the full snippet.'
+                    )
 
             await wait_for_deletion(
```

Comment:
Only saw this after I pressed the submit review button, but `wait_for_deletion()` only accepts messages that are in guilds. Although, I don't think we should really support code snippets in DMs anyways.

Reply:
> I don't think we should really support code snippets in DMs anyways.

I think it's nice to support as many features in DMs as we can. It's relatively easy to support code snippets in DMs, so I think we should.

> `wait_for_deletion()` only accepts messages that are in guilds.

Thanks, have fixed :+1:

Reply:
> I think it's nice to support as many features in DMs as we can. It's relatively easy to support code snippets in DMs, so I think we should.

I disagree with this, and I believe that’s the general view of the core dev team as well. There is a very, very clear cost (this entire PR) to supporting these features in DMs, which doesn’t seem to balance against the benefits. You could always just… run this feature in bot-spam.

Reply:
Eh, I guess.

I've fixed this now though so we might as well merge it 🤷.

Reply:
I agree with Scale here, we don't want to support things in DMs with the bot. I think the correct solution to this sentry issue is to just add the below conditional to the top of the event listener.
```py
if not message.guild:
    return
```

Reply:
Done in https://github.com/python-discord/bot/pull/2086/commits/95106690fa32d1cc5578e018ee7c75857d41d8a5.